### PR TITLE
Shorten discoverable name.

### PR DESCRIPTION
**Pull request by anwar6953**
https://issues.cask.co/browse/CDAP-15303
https://builds.cask.co/browse/CDAP-RUT1674-8
https://builds.cask.co/browse/IT-ITM7-14 (single test case)

### CODE REVIEW

File: cdap-common/src/test/java/io/cdap/cdap/common/guice/ZKDiscoveryModuleTest.java [hunk 1]
```diff
@@ -122,10 +122,10 @@ public void testProgramDiscovery() {
       // Register a service using the twill ZKClient. This is to simulate how a user Service program register
       ProgramId programId = NamespaceId.DEFAULT.app("app").service("service");
       String twillNamespace = injector.getInstance(CConfiguration.class).get(Constants.CFG_TWILL_ZK_NAMESPACE);
-      ZKClient twillZKClient = ZKClients.namespace(zkClient,
-                                                   twillNamespace + "/" + TwillAppNames.toTwillAppName(programId));
 
       String discoverableName = ServiceDiscoverable.getName(programId);
+      ZKClient twillZKClient = ZKClients.namespace(zkClient,
+                                                   twillNamespace + "/" + discoverableName);
```
Comment (anwar6953) [inline cdap-common/src/test/java/io/cdap/cdap/common/guice/ZKDiscoveryModuleTest.java]: this test case fails without this change... the zkclient's namespace is impacting the discovery.

File: cdap-common/src/main/java/io/cdap/cdap/common/service/ServiceDiscoverable.java [hunk 2]
```diff
@@ -43,32 +45,41 @@
...
+  }
+
   public static String getName(ProgramId programId) {
     return getName(programId.getNamespace(), programId.getApplication(), programId.getType(), programId.getProgram());
   }
 
   public static String getName(String namespaceId, String appId, ProgramType programType, String programName) {
-    if (!USER_SERVICE_TYPES.contains(programType)) {
-      throw new IllegalArgumentException("Program type should be one of " + USER_SERVICE_TYPES);
+    if (!SHORT_NAME_TO_PROGRAM_TYPE_MAPPING.containsValue(programType)) {
```
Comment (chtyim) [inline cdap-common/src/main/java/io/cdap/cdap/common/service/ServiceDiscoverable.java]: Why need to change this? The `programType` should be in the `USER_SERVICE_TYPES`, right?

Comment (anwar6953) [reply cdap-common/src/main/java/io/cdap/cdap/common/service/ServiceDiscoverable.java]: we need a reverse mapping.
Lookup the ProgramType based on the short form.
We could have a switch statement instead...

Comment (chtyim) [reply cdap-common/src/main/java/io/cdap/cdap/common/service/ServiceDiscoverable.java]: you are doing a `SHORT_NAME_TO_PROGRAM_TYPE_MAPPING.containsValue(programType)`, which is equivalent to doing `USER_SERVICE_TYPES.contains(programType)`, isn't it?

Comment (anwar6953) [reply cdap-common/src/main/java/io/cdap/cdap/common/service/ServiceDiscoverable.java]: Ya but I had to change the data structure from set to map for the usage on line 76.

Comment (chtyim) [reply cdap-common/src/main/java/io/cdap/cdap/common/service/ServiceDiscoverable.java]: But you didn't delete the `USER_SERVICE_TYPES` constant, so why not just use it here?

Comment (anwar6953) [reply cdap-common/src/main/java/io/cdap/cdap/common/service/ServiceDiscoverable.java]: I thought I had deleted the map.
I reverted this change now.

**Comment by anwar6953** [comment]
Addressed comments, please take another look.

**Review by chtyim — APPROVED** [review]

**Comment by anwar6953** [comment]
Squashed commits.

**Review by chtyim — APPROVED** [review]

**Pull request merged**